src/main/java/org/jenkins_ci/plugins/run_condition/core/CauseCondition.java

```diff
@@ -66,15 +68,17 @@ public boolean isCausedBy(String className) {
 
         // if XTrigger plugin is installed:
         // file change
-        FS_CAUSE("org.jenkinsci.plugins.fstrigger.FSTriggerCause", "FSTrigger"),
+        FS_CAUSE(XTriggerCause.class, "FSTrigger"),
```

Comment:
It's been awhile, but I believe it was because the XTrigger integration
didn't actually work with the strings. They were always an XTriggerCause
class no matter what triggered it.

On Wed, Mar 30, 2016 at 2:38 AM Dominik Bartholdi <notifications@github.com>
wrote:

> In
> src/main/java/org/jenkins_ci/plugins/run_condition/core/CauseCondition.java
> <https://github.com/jenkinsci/run-condition-plugin/pull/9#discussion_r57842590>
> :
>
> > @@ -66,15 +68,17 @@ public boolean isCausedBy(String className) {
> >
> >          // if XTrigger plugin is installed:
> >          // file change
> > -        FS_CAUSE("org.jenkinsci.plugins.fstrigger.FSTriggerCause", "FSTrigger"),
> > +        FS_CAUSE(XTriggerCause.class, "FSTrigger"),
>
> why did you change this to use the class instead of the class name as a
> String? now the plugin has a dependency to the xtrigger plugin and
> everyone using this plugin needs to install that plugin too. Please change
> this back to use the full qualified name of the class as a String
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/run-condition-plugin/pull/9/files/929a10d1248b9224b895ff6f7eccfa6d9ed7dc68#r57842590>
>
-- 
Typed on a tiny keyboard.